src/nvim/shada.c

```diff
@@ -2402,6 +2413,46 @@ static inline ShaDaWriteResult shada_read_when_writing(
   return ret;
 }
 
+/// Initializes registers for writing to the ShaDa file
+///
+/// @param[in]  wms  The WriteMergerState used when writing.
+/// @param[in]  max_reg_lines The maximum number of register lines.
+static void shada_initialize_registers(WriteMergerState *const wms,
+                                       int max_reg_lines) {
```

Comment:
When I add `FUNC_ATTR_ALWAYS_INLINE` the compiler gives a warning ` always_inline function might not be inlinable [-Wattributes]`